### PR TITLE
apps in memory; fixed save new app test

### PR DESCRIPTION
**Pull request by scottnath**
A break up of my larger PR.

This PR adds `applications` to the express app memory, and also adds it to the request object. System changes the object on applications change

---

`DCO 1.1 Signed-off-by: Scott Nath <github@scottnath.com>`

### CODE REVIEW

File: lib/init/settings.js [hunk 1]
```diff
@@ -24,6 +24,20 @@ module.exports = (app) => {
   }).then(merged => {
     app.set('applications-merged', merged[0]);
 
+    return database('applications').select('*').orderBy('name', 'DESC').then(rows => {
```
Comment (Snugug) [inline lib/init/settings.js]: I fell in to this trap with APIs. We can't grab these only during initialization, we need to grab these when they're needed. Otherwise, if they change _after_ initialization, we won't have the updated list.

File: lib/init.js [hunk 2]
```diff
@@ -22,14 +22,14 @@ module.exports = (application) => {
...
   }).then(app => {
-    return schedule(app);
-  }).then(app => {
     return passport(app);
   }).then(app => {
     return acl(app);
   }).then(app => {
     return settings(app);
   }).then(app => {
+    return schedule(app);
```
Comment (Snugug) [inline lib/init.js]: Why did you move schedule?

File: lib/applications/routes.js [hunk 3]
```diff
@@ -299,10 +291,13 @@ const save = (req, res) => {
     data['client-id'] = uuid.v4();
     data['client-secret'] = uuid.v4();
 
-    return database('applications').insert(data).returning('id').then((newid) => {
+    return database('applications').insert(data).returning('*').then((added) => {
       _.unset(req.session, 'form.applications.save');
 
-      res.redirect(`/${config.applications.base}/${newid}`);
+      // add this new app to the request's apps object
+      req.applications.apps.push(added[0]);
```
Comment (Snugug) [inline lib/applications/routes.js]: Does this actually update `app.get('applications')`?

File: lib/applications/routes.js [hunk 4]
```diff
@@ -285,10 +269,18 @@ const save = (req, res) => {
...
         _.unset(req.session, 'form.applications.edit');
         _.unset(req.session, 'form.applications.save');
 
+        // find this app in the request apps object
+        const index = req.applications.apps.findIndex(app => {
+          return app.id === id;
+        });
+
+        // we WANT to update this app in the request's apps object
+        req.applications.apps[index] = updated[0]; // eslint-disable-line no-param-reassign
```
Comment (Snugug) [inline lib/applications/routes.js]: Can we use lodash `_.set` instead of disabling our linter?

**Comment by Snugug** [comment]
Discussed, going to convert from `app.get` and direct `req` manipulation in routes to middleware for the API routes

File: lib/init.js (hunk 2 above)
Comment (Snugug) [inline lib/init.js]: 🐗 

(no reason, just a fancy emoji I hadn't seen before)

File: lib/init/settings.js (hunk 1 above)
Comment (scottnath) [inline lib/init/settings.js]: I've changed this now and I'm updating the express settings on change of apps.
